### PR TITLE
add options to choose post types to be shortened

### PR DESCRIPTION
**Pull request by finagle29**
I think I saw an ask requesting to be able to shorten audio posts, so I figured why not have the option to choose exactly which types of posts are shortened

### CODE REVIEW

**Comment by finagle29** [comment]
fixes #330

File: Extensions/shorten_posts.js [hunk 1]
```diff
@@ -21,10 +21,40 @@ XKit.extensions.shorten_posts = new Object({
 			text: "When to shorten posts",
 			type: "separator"
 		},
-		only_text: {
```
Comment (bvtsang) [inline Extensions/shorten_posts.js]: Do we want to remove the option for the user to remove long text posts? What if they want to shorten only photosets? Do we want to support that use case?

**Comment by finagle29** [comment]
Long text posts are always shortened, but I can add an option so that there is full customization.
It'd clean up the code a bit too.

File: Extensions/shorten_posts.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 //* TITLE Shorten Posts **//
-//* VERSION 0.2.1 **//
+//* VERSION 0.2.2 **//
 //* DESCRIPTION Makes scrolling easier **//
-//* DETAILS This extension shortens long posts, so if you are interested, you can just click on Show Full Post button to see it all, or scroll down if you are not interested. Useful for screens where long posts take a lot of space, and making it hard to scroll down.<br><br>By default, this extension only shortens text posts. You can toggle the setting to let it shorten the photo posts too. (This will 'cut off' long, vertical posts.) **//
+//* DETAILS This extension shortens long posts, so if you are interested, you can just click on Show Full Post button to see it all, or scroll down if you are not interested. Useful for screens where long posts take a lot of space, and making it hard to scroll down.<br><br>By default, this extension shortens text posts. You can toggle settings to let it shorten other types of posts, as well as not shorten text posts. (This will 'cut off' long, vertical posts.) **//
```
Comment (dlmarquis) [inline Extensions/shorten_posts.js]: I think "...toggle the settings to choose which post types to shorten..." would be cleaner for the last bit of the desc.

Comment (finagle29) [inline Extensions/shorten_posts.js]: i completely agree

File: Extensions/shorten_posts.js [hunk 3]
```diff
@@ -73,16 +108,42 @@ XKit.extensions.shorten_posts = new Object({
...
+				dont_return = true;
+			}
+			if (XKit.extensions.shorten_posts.preferences.photos_too.value
+				&& ($(this).hasClass("is_photo") || $(this).hasClass("is_photoset"))) {
+				dont_return = true;
+			}
+			if (XKit.extensions.shorten_posts.preferences.audio_too.value
+				&& $(this).hasClass("is_audio")) {
+				dont_return = true;
 				}
```
Comment (bvtsang) [inline Extensions/shorten_posts.js]: Nitpick: indentation for this curly brace is a bit off

**Comment by finagle29** [comment]
are there any other small fixes/nitpick-type things that i should fix?

**Comment by bvtsang** [comment]
LGTM

@homu r+

**Pull request merged**